### ISSUE TITLE
JSON filename should default to input (or output) filename

### ISSUE THREAD

**Issue by morewry** [tivac/modular-css#459]
I think that it should be possible to generate JSON data for CSS files [without having to provide a specific name for the file](https://github.com/tivac/modular-css/blob/master/packages/postcss/postcss.js#L38). 

If the `json` option is set to `true`, I think the JSON file should be generated with the same name as the input file. So this configuration with an input file `input.css` should generate `input.json` (or `input.css.json`).

```json
{
  "postcss-modular-css": {
    "json": true
  }
}
```

Another, similar option that would similarly accomplish the desired result would be to emulate Webpack, which has a syntax for naming an output file the same name as input file, e.g. `[name].js`

```json
{
  "postcss-modular-css": {
    "json": "[name].json"
  }
}
```

The use case that inspired this request is [postcss-modular-css](https://github.com/tivac/modular-css/tree/master/packages/postcss) with [postcss-cli](https://github.com/postcss/postcss-cli).

There are two relevant examples in the documentation, one with a postcss configuration file (`postcss --config postcss.json input.css`) and one without (`postcss --use modular-css/postcss input.css`):

###### postcss.json

```json
{
  "output" : "out.css",
  "postcss-modular-css": {
    "json" : "./path/to/output.json"
  }
}
```

But `postcss-cli` supports globs for input (and output directories for multiple files), e.g. `postcss ./src/*.css -d ./dist/`. So it would be nice if you could use a configuration like this with that command:

###### .postcssrc.json
```json
{
  "plugins": {
    "postcss-modular-css": {
      "json": true
    }
  }
}
```

**Comment by tivac** [comment]
Sounds reasonable to me! Rollup/Webpack handle this sort of thing natively but postcss doesn't provide much. I'd rather not get into supporting string replacements mostly because it's more code and seems less useful to me since it'll always just be one or two replacements.

So how about if `json` is truthy but not a string we'll use the `options.to` value provided by postcss as the base filename for the output. Should it always append `.json` or should it replace whatever the extension is with `.json`? The difference is small but I don't have a horse in this race so I appreciate any opinions!

**Comment by morewry** [comment]
Yep, that sounds great.

As to the filename, I don't have strong feelings, but I _slightly_ prefer the simpler output file name that would result by replacing the the `options.to` extension with `.json`.

Since you make it easy, I opened a PR. Feel free to merge or not as suits you :-)

**Comment by tivac** [comment]
This was released in `14.2.0`
